src/actions/sources.js

```diff
@@ -40,37 +41,52 @@ import {
   removeSourcesFromTabList,
   removeSourceFromTabList,
   getTextSearchQuery,
-  getActiveSearch
+  getActiveSearch,
+  getGeneratedSource
 } from "../selectors";
 
 import type { Source, Location } from "../types";
 import type { ThunkArgs } from "./types";
-import type { State } from "../reducers/types";
 
 // If a request has been made to show this source, go ahead and
 // select it.
-async function checkSelectedSource(state: State, dispatch, source) {
-  const pendingLocation = getPendingSelectedLocation(state);
+function checkSelectedSource(source) {
+  return async ({ dispatch, getState }: ThunkArgs) => {
```

Comment:
I made both `checkSelectedSource` and `checkPendingBreakpoints` action creators because that simplified things / made them more consistent.